### ISSUE TITLE
Balance Issue

### ISSUE THREAD

**Issue by Chapof1** [HypsyNZ/Binance-Trader#1]
When I try to sell it says my balance is insufficient, and it is, because of BINANCE FEES, so the bot keeps selling higher amount than I have. Any idea how to fix this?

**Comment by HypsyNZ** [comment]
Thanks for reporting this, I'll fix it today some time.

In the mean time if you want to use the `Scraper` you could use it on the [`No Fees`](https://www.binance.com/en/support/announcement/binance-launches-zero-fee-bitcoin-trading-10435147c55d4a40b64fcbf43cb46329) pairs that it was designed for

**Comment by HypsyNZ** [comment]
Thanks for reporting this, Its quite complicated to consider fees because every coin has a minimum quantity step so there may be `Dust` left over after each sell, Do you think this is acceptable?

As an example (not real math), If you were to use the `Scraper` on `Doge` coin and placed `$12` orders there would be

If the `DOGE` price was `0.09448` and you bought `123` coins, It would cost `11.62104` and the fee would be `0.02324208` this means when the scraper sells it can only sell `122` and you would have `7 cents` worth of `DOGE` left over...

It would also mean there is a `Minimum PnL` which is the `Fee x 2`

----

In the mean time if you want to use the `Scraper` you could use it on the [`No Fees`](https://www.binance.com/en/support/announcement/binance-launches-zero-fee-bitcoin-trading-10435147c55d4a40b64fcbf43cb46329) pairs that it was designed for

**Comment by HypsyNZ** [comment]
Thanks for reporting this, Its quite complicated to consider fees because every coin has a minimum quantity step so there may be `Dust` left over after each sell, Do you think this is acceptable?

If the `DOGE` price was `0.09448` and you bought `123` coins, It would cost `11.62104` and the fee would be `0.02324208` this means when the scraper sells it can only sell `122` and you would have `7 cents` worth of `DOGE` left over...

It would also mean there is a `Minimum PnL` which is the `Fee x 2`

----

In the mean time if you want to use the `Scraper` you could use it on the [`No Fees`](https://www.binance.com/en/support/announcement/binance-launches-zero-fee-bitcoin-trading-10435147c55d4a40b64fcbf43cb46329) pairs that it was designed for

**Comment by HypsyNZ** [comment]
What do you think of an option to use the `Dust` to pay `Interest` if you are borrowing?

**Comment by Chapof1** [comment]
> What do you think of an option to use the `Dust` to pay `Interest` if you are borrowing?

Haven't tried it yet, but would be good to try to fix that balance issue because there would be no utility using Scraper then, unless you use the ''no fees'' trading pars, but that only includes BTC markets...

**Comment by Chapof1** [comment]
> Thanks for reporting this, Its quite complicated to consider fees because every coin has a minimum quantity step so there may be `Dust` left over after each sell, Do you think this is acceptable?
> 
> If the `DOGE` price was `0.09448` and you bought `123` coins, It would cost `11.62104` and the fee would be `0.02324208` this means when the scraper sells it can only sell `122` and you would have `0.07123792 USDT` worth of `DOGE` left over...
> 
> It would also mean there is a `Minimum PnL` which is the `Fee x 2`
> 
> In the mean time if you want to use the `Scraper` you could use it on the [`No Fees`](https://www.binance.com/en/support/announcement/binance-launches-zero-fee-bitcoin-trading-10435147c55d4a40b64fcbf43cb46329) pairs that it was designed for

Yes it could be acceptable if the ''dust'' is not big enough to cover profit taken from the scraper, I think

**Comment by HypsyNZ** [comment]
> unless you use the ''no fees'' trading pars

Yes this is what it was designed for

> if the ''dust'' is not big enough to cover profit

It will take a bit of effort to make sure the trade is net profit, because there is the `Fee` and the `Dust` to consider but it should be workable.

I could keep track of the dust and then increase the quantity and sell it on one the later orders, but that assumes the user is going to keep it running for x period of time when you can randomly start/stop the scraper so there is a lot to think about.

**Comment by HypsyNZ** [comment]
I have released a new version that disables the `Scraper` on symbols that have fees

If you are going to use a previous version to scrape coins that have fees you need to increase the `Scrap Percent` to something that covers the `Fees` and `Dust` and have a little bit of extra balance in your cross account.

For symbols with `Fees` the `Scrap Percent` is probably too low

**Comment by HypsyNZ** [comment]
You can use the `Scraper` for symbols that have fees but this issue is still not resolved.
